src/nodes/composite/predicate.ts

```diff
@@ -278,28 +279,26 @@ export const assertAllowsConstraint = (
         case "morph":
             return
         default:
-            throwInternalError(`Unexpxected rule kind '${node.kind}'`)
+            throwInternalError(
+                `Unexpected rule kind '${(node as ConstraintNode).kind}'`
+            )
     }
 }
 
 const assertValidLimit = (bounds: Range, boundType: "number" | "Date") => {
     for (const index in bounds) {
```

Comment:
Typically you don't want to use `for...in` for an array unless you need to iterate over non-numeric keys